### DISCUSSION TITLE
Idea for stableness of minification and tranpilation

### DISCUSSION THREAD

**Discussion by kdy1** [swc-project/swc#8075] [General]
> For library authors, if you provide a repository using @swc/core and some description yaml file that describes the way to run tests, we will run them before publishing new versions, to ensure that the new PR never introduces regressions

Will this work and library authors join?

**Comment by SukkaW** [comment]
Hello, the author and the maintainer of the [`rollup-plugin-swc3`](https://www.npmjs.com/package/rollup-plugin-swc3) here. Sign me in.

**Reply by SukkaW** [reply]
I am quite interested in this initiative. Just months ago, `swc@1.3.76`'s behavior changes just break `rollup-plugin-swc3`: 

- Issue: https://github.com/SukkaW/rollup-plugin-swc/issues/41
- Fix (part 1): https://github.com/SukkaW/rollup-plugin-swc/pull/44
- Fix (part 2): https://github.com/SukkaW/rollup-plugin-swc/commit/20de029de538a7ae5e03e86d7a5bf144e25eb487 

I would appreciate swc notifying library authors of behavior changes like this.

**Reply by kdy1** [reply]
Can you take a look at https://github.com/swc-project/swc-ecosystem-ci/actions/runs/6492356302/job/17631158201 ?

**Reply by SukkaW** [reply]
@kdy1 Sure!

**Reply by SukkaW** [reply]
@kdy1 I have setup `swc-ecosystem-ci` on my local machine with `rollup-plugin-swc3`. I noticed that `@swc/types` is missing in the setup, causing the build issue.

**Reply by kdy1** [reply]
Is it a bug of `swc-ecosystem-ci`?
If not, can you update `rollup-plugin-swc3`? I think it should have `@swc/types` in `devDependencies`

**Reply by kdy1** [reply]
<img width="1297" alt="image" src="https://github.com/swc-project/swc/assets/29931815/8a582d36-ba22-43ab-8553-a45c891457b6">

Thank you! It works!

https://github.com/swc-project/swc-ecosystem-ci/actions/runs/6504759902

**Comment by DanRibbens** [comment]
We have pinned swc for Payload because of a bug that was released a few months ago. I could provide test info if you're interested in adding https://github.com/payloadcms/payload to your CI.

**Reply by kdy1** [reply]
I'm definitely interested

**Reply by kdy1** [reply]
Can you send a PR to https://github.com/swc-project/swc-ecosystem-ci ?
